enhancements/baremetal/bootstrap-external-ip.md

```diff
@@ -0,0 +1,130 @@
+---
+title: bootstrap-external-ip
+authors:
+  - "@honza"
+reviewers:
+  - "@hardys"
+  - "@dtantsur"
+  - "@dhellmann"
+  - TBD
+approvers:
+  - TBD
+api-approvers:
+  - TBD
+creation-date: 2022-03-23
+last-updated: 2022-03-23
+tracking-link:
+  - https://issues.redhat.com/browse/METAL-175
+see-also:
+  - https://bugzilla.redhat.com/show_bug.cgi?id=2048600#c1
+replaces:
+superseded-by:
+---
+
+# Bootstrap External IP
+
+## Summary
+
+When installing a bare metal IPI cluster, you can use the `networkConfig` field
+in the `install-config.yaml` file to configure the control plane network
+interfaces for the cluster hosts.  However, currently, you cannot configure the
+bootstrap VM networking using the same means.
+
+## Motivation
+
+In environments where no DHCP server is running, the bootstrap VM will not get
+an IP address on the control plane NIC causing cluster installation to fail.
+The user can work around this by modifying the bootstrap ignition file; however,
+this isn't a friendly experience.
+
+### Goals
+
+- Support cluster installation in environments without a running DHCP server
+- Improve the level of customization of bootstrap VM networking
+- Improve user-friendliness when specifying a static IP for the bootstrap VM
+
+### Non-Goals
+
+- Provide means for configuring every aspect of bootstrap networking
+
+## Proposal
+
+We will add a new field, `bootstrapExternalIP`, to allow for further
+customization of bootstrap networking.  We will use this IP address to generate
+static configuration in the installer.
+
+### User Stories
+
+- As a user, I want to be able to deploy an IPI bare metal cluster in an
+  environment where no DHCP cluster is running
+- As a user, I want to specify a static IP address for the bootstrap VM
+
+### API Extensions
+
+We will add a new field to the `install-config.yaml` file called
+`bootstrapExternalIP`.  This is similar to the existing
+`bootstrapProvisioningIP` field.
```

Comment:
Do we need to add additional validations for this new field? If so, what would they be?

Reply:
I think just verifying it's in the correct IP format should be fine.

Reply:
I think just verifying it's in the correct IP format should be fine.

Reply:
We should validate [similar to other fields](https://github.com/openshift/installer/blob/master/pkg/types/baremetal/validation/platform.go#L410..L414) e.g

* IP syntax is valid
* IP is in the MachineNetwork
* IP doesn't conflict with other IPs (e.g API/Ingress VIP IPs)

Reply:
Thanks! I was wondering if there is some correlation with the `networkConfig` field. Since nothing is mentioned in the doc about it, I am assuming that they can occur independently. We could configure `bootstrapExternalStaticIP`, and
`bootstrapExternalGateway` irrespective of whether the `networkConfig` field is used to configure StaticIPs on hosts within the cluster.